homeassistant/components/modbus/base_platform.py

```diff
@@ -76,7 +77,7 @@ class BasePlatform(Entity):
     def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
         """Initialize the Modbus binary sensor."""
         self._hub = hub
-        self._slave = entry.get(CONF_SLAVE, 0)
+        self._slave = entry.get(CONF_SLAVE, entry.get(CONF_DEVICE_ADDRESS, 0))
```

Comment:
```suggestion
        self._slave = entry.get(CONF_SLAVE) or entry.get(CONF_DEVICE_ADDRESS, 0)
```
Will only evaluate `entry.get(CONF_DEVICE_ADDRESS, 0)` when so slave is set.

Reply:
Which is correct, since they are defined as being exclusive in __init__.py

Reply:
I am not saying the statement is in correct, but only that is more efficient using `or`

Reply:
Changed to your way, I have no strong feelings about "or" or the other.

Reply:
Thnx!